designer/new_dialog.py

```diff
@@ -77,6 +77,8 @@ def __init__(self, **kwargs):
         super(NewProjectDialog, self).__init__(**kwargs)
         item_strings = list(NEW_PROJECTS.keys())
         item_strings.sort()
+        self._keyboard = Window.request_keyboard(self._keyboard_closed, self)
```

Comment:
Use the `Window.bind(on_key_down..` here. It's already being used by shortcut manager so we can have more control.